src/commands/CommandEditAlertParserForm.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react'
+import { PropTypes } from '@dhis2/prop-types'
+import {
+    Button,
+    ReactFinalForm,
+    FinalForm,
+    NoticeBox,
+    CenteredContent,
+    CircularLoader,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import { useDataQuery, useDataEngine } from '@dhis2/app-runtime'
+import { useHistory } from 'react-router'
+
+import { SMS_COMMAND_LIST_PATH } from '../views'
+import i18n from '../locales'
+import { FormRow } from '../forms'
+import { FieldCommandName } from './FieldCommandName'
+import { FieldCommandParser } from './FieldCommandParser'
+import { FieldCommandConfirmMessage } from './FieldCommandConfirmMessage'
+
+const { Form } = ReactFinalForm
+const { FORM_ERROR } = FinalForm
+
+const query = {
+    smsCommand: {
+        resource: 'smsCommands',
+        id: ({ id }) => id,
+        params: {
+            fields: [
+                'name',
+                'parserType',
+                'receivedMessage',
+                'userGroup[name]',
+            ],
+            paging: 'false',
+        },
+    },
+}
+
+const mutation = {
+    resource: 'smsCommands',
+    type: 'update',
+    partial: true,
+    id: ({ id }) => id,
+    data: ({ command }) => command,
+}
+
+export const CommandEditAlertParserForm = ({ id }) => {
+    const history = useHistory()
+    const engine = useDataEngine()
+    const updateCommand = command =>
+        engine
+            .mutate(mutation, { variables: { command, id } })
+            .then(() => {
+                history.push(SMS_COMMAND_LIST_PATH)
+            })
+            .catch(error => {
+                const isValidationError = error.type === 'access'
```

Comment:
I'm not that familiar with the new error codes the webapi is producing, but at first glance this doesn't make sense to me.... If the type is `access` I would assume this error is the result of some authentication issue, and not really to do with validation (i.e. sth being wrong with the data)...

Reply:
Yeah I agree. This is what Austin said to use to catch validation type errors. I'd have to look into the data engine source, but it made sense to me at the time for the range of error codes associated with `access`.